public/app/partials/confirm_modal.html

```diff
@@ -27,7 +26,7 @@ <h2 class="modal-header-title">
 
 		<div class="confirm-modal-buttons">
 			<button ng-show="onAltAction" type="button" class="btn btn-primary" ng-click="dismiss();onAltAction();">{{altActionText}}</button>
-			<button type="button" class="btn btn-danger" ng-click="onConfirm();dismiss();" ng-disabled="!confirmTextValid" give-focus="true">{{yesText}}</button>
+			<button ng-show="onConfirm" type="button" class="btn btn-danger" ng-click="onConfirm();dismiss();" ng-disabled="!confirmTextValid" give-focus="true">{{yesText}}</button>
```

Comment:
Hmm isn't this also a breaking change. With this change the modal is not automatically dismissed?

Reply:
Not sure. If there is no `onConfirm` it will not show confirm button, just cancel button. Before that not defining `onConfirm` would result in error as the button would be there but `onConfirm` would be undefined so I assume nobody used it like that so far. So it would be a new behaviour but not a breaking one and all the usages should still work in the same way.

Reply:
I mean if `onConfirm` is set like in this code:
https://github.com/grafana/grafana/blob/9b393d8f6440bbbbe239149c34f33e7f244305f5/public/app/core/components/manage_dashboards/manage_dashboards.ts#L204-L213

Will the modal still be automatically closed now after `deleteFoldersAndDashboards` have been executed like it currently works?

Reply:
Yes it should. The template is `ng-click="onConfirm();dismiss();"` which looks kinda weird but will do both `onConfirm()` and `dismiss()` afterwards no matter what you do inside `onCofirm`. So only change is that with 
```
appEvents.emit('confirm-modal', { 
   title: 'Delete', 
   text: text, 
   text2: text2, 
   icon: 'fa-trash', 
   onConfirm: undefined
 }); 
```
It will only show `Cancel` button basically acting as an `alert()`. I would do a separate kind of modal (like `alert-modal`) for that ideally, but as this is in angular I did not want to spend too much time on that.

Reply:
Sorry my fault. Could swear it looked like you've changed `ng-click="onConfirm();dismiss();"` to `ng-click="onConfirm();"` :man_facepalming: